### ISSUE TITLE
On Mi Mix Fold both screens are together on

### ISSUE THREAD

**Issue by LeOS-GSI** [phhusson/treble_experimentations#1955]
On Mix Fold, inside and outside screen are on together, but outside screen should only be on when device is fold. 

Have tried to implement all Mix Fold hardware overlays, but overlays won't be available in build or couldn't be activated.

**Comment by phhusson** [comment]
1. Please run `getevent` and see what event happens when you close/open the screen
2. Get the result of `adb shell dumpsys display`

**Comment by phhusson** [comment]
There is a WIP Moto Razr support here: https://github.com/phhusson/treble_app/commit/d29c57504ff69b5723216753d8db62fddfc8c444#diff-713f830fcdccf335b479464d356cb3e8acf49c5632ab51d4d1c10240aa0dd4dbR87
It shows how to turn displays on/off displays independantly

**Comment by phhusson** [comment]
Also add the result of `adb shell dumpsys input` for touchscreen, and explain what's the behaviour of external touchscreen

**Comment by LeOS-GSI** [comment]
Attached the dump and event files. The external/outside screen is showing the same as the inside screen, but only in small size (not over the whole size of outer screen size). Touch isn't working on outer screen.
[uevent.txt](https://github.com/phhusson/treble_experimentations/files/6988375/uevent.txt)
[getevent.txt](https://github.com/phhusson/treble_experimentations/files/6988376/getevent.txt)
[dumpsys_input.txt](https://github.com/phhusson/treble_experimentations/files/6988377/dumpsys_input.txt)
[dumpsys_display.txt](https://github.com/phhusson/treble_experimentations/files/6988378/dumpsys_display.txt)

**Comment by phhusson** [comment]
Interestingly, external's screen touchscreen fts1 is simply disabled:
`Enabled: false`
I don't even know how an OEM can do that.
The touchscreens aren't assigned to displays, so it'll be mixed up, but again there is code to do that in moto razr poc code: https://github.com/phhusson/treble_app/commit/d29c57504ff69b5723216753d8db62fddfc8c444#diff-713f830fcdccf335b479464d356cb3e8acf49c5632ab51d4d1c10240aa0dd4dbR43

**Comment by LeOS-GSI** [comment]
OK, will check the razr code.

here the next logs and a picture of closed screen
[uevent_when_closing.txt](https://github.com/phhusson/treble_experimentations/files/6988418/uevent_when_closing.txt)
[uevent_closed.txt](https://github.com/phhusson/treble_experimentations/files/6988419/uevent_closed.txt)
[getevent_closed.txt](https://github.com/phhusson/treble_experimentations/files/6988420/getevent_closed.txt)
[dumpsys_display_closed.txt](https://github.com/phhusson/treble_experimentations/files/6988421/dumpsys_display_closed.txt)
[dumpsys_display_open.txt](https://github.com/phhusson/treble_experimentations/files/6988422/dumpsys_display_open.txt)
[getevent_open_device.txt](https://github.com/phhusson/treble_experimentations/files/6988423/getevent_open_device.txt)
![Fold_open](https://user-images.githubusercontent.com/2726574/129484295-a82c34b3-a28c-4de1-9f0b-6170ca448fc4.jpg)

**Comment by LeOS-GSI** [comment]
> Interestingly, external's screen touchscreen fts1 is simply disabled:
> `Enabled: false`
> I don't even know how an OEM can do that.
> The touchscreens aren't assigned to displays, so it'll be mixed up, but again there is code to do that in moto razr poc code: [phhusson/treble_app@d29c575#diff-713f830fcdccf335b479464d356cb3e8acf49c5632ab51d4d1c10240aa0dd4dbR43](https://github.com/phhusson/treble_app/commit/d29c57504ff69b5723216753d8db62fddfc8c444#diff-713f830fcdccf335b479464d356cb3e8acf49c5632ab51d4d1c10240aa0dd4dbR43)

I have rebuild the app with the changes and the Razr.kt and build a new GSI. But nothing changed. Have I to rename to 'Fold' or something else ?

**Comment by phhusson** [comment]
Yes, you have to do much more than that. And renaming the file won't change anything. 

First you need to change the fingerprint match L35.

So your vendor has /vendor/etc/input-port-associations.xml ; so display/touchscreen mapping should work. Possibly, fts1 is disabled because the secondary display is currently just a copy of the primary one.

Please go in developer settings, and enable "Force desktop mode". You'll probably need to reboot after that. Once this is done, you should have a different launcher on the external screen compared to internal. Do you confirm? And touchscreen should work on external. Do you confirm?

**Comment by LeOS-GSI** [comment]
Ok, now outside screen has different layout, but touch isn't working.

**Comment by LeOS-GSI** [comment]
So, after compiled a new GSI with new treble app and attached fold.kt same as before. outside screen is showing a different desktop but no touch working. inside screen is OK.
![fold_ouside1](https://user-images.githubusercontent.com/2726574/129553992-ca76b78e-f7a7-4bd1-9dec-d03c7cdd4e06.jpg)
[MiFold.kt.txt](https://github.com/phhusson/treble_experimentations/files/6991865/MiFold.kt.txt)

**Comment by phhusson** [comment]
you don't need to rebuild GSI to test new treble app, you can just install
it like any other app

**Comment by julianbraha** [comment]
I am still having this issue on the new 12L gsi. Can I help in some way @phhusson?
